framework/Samplers/AdaptiveMonteCarlo.py

```diff
@@ -0,0 +1,226 @@
+                                  'prefix':prefix,
+                                  'tol':tol
+                                  })
+          else:
+            self.raiseAWarning('Unrecognized convergence node "',tag,'" has been ignored!')
+        assert (len(self.toDo)>0), self.raiseAnError(IOError, ' No target have been assigned to convergence node')
+      elif child.getName() == "initialSeed":
+        self.initSeed = child.value
+    for metric, infos in self.toDo.items():
+      steMetric = metric + '_ste'
+      if steMetric in self.statErVals:
+        for info in infos:
+          prefix = info['prefix']
+          for target in info['targets']:
+            metaVar = prefix + '_ste_' + target
+            self.tolerance[metaVar] = info['tol']
+
+  def localInitialize(self,solutionExport=None):
+    """
+      Will perform all initialization specific to this Sampler. For instance,
+      creating an empty container to hold the identified surface points, error
+      checking the optionally provided solution export and other preset values,
+      and initializing the limit surface Post-Processor used by this sampler.
+      @ In, solutionExport, DataObjects, optional, a PointSet to hold the solution
+      @ Out, None
+    """
+    self.converged        = False
+    self.basicStatPP   = BasicStatistics(self.messageHandler)
+    if 'TargetEvaluation' in self.assemblerDict.keys():
+      self.lastOutput = self.assemblerDict['TargetEvaluation'][0][3]
+    self.solutionExport    = solutionExport
+    # check if solutionExport is actually a "DataObjects" type "PointSet"
+    if solutionExport.type != "PointSet":
+      self.raiseAnError(IOError,'solutionExport type is not a PointSet. Got '+ solutionExport.type +'!')
+
+    self.basicStatPP.what = self.toDo.keys()
+    self.basicStatPP.toDo = self.toDo
+    self.basicStatPP.initialize({'WorkingDir':None},[self.lastOutput],{'Output':[]})
+    self.raiseADebug('Initialization done')
+
+  ###############
+  # Run Methods #
+  ###############
+
+  def localFinalizeActualSampling(self,jobObject,model,myInput):
+    """
+      General function (available to all samplers) that finalize the sampling
+      calculation just ended. In this case, The function is aimed to check if
+      all the batch calculations have been performed
+      @ In, jobObject, instance, an instance of a JobHandler
+      @ In, model, model instance, it is the instance of a RAVEN model
+      @ In, myInput, list, the generating input
+      @ Out, None
+    """
+    if self.counter>1:
+      output = self.basicStatPP.run(self.lastOutput)
+      output['solutionUpdate'] = np.asarray([self.counter - 1])
+      self.solutionExport.addRealization(output)
+      self.checkConvergence(output)
+
+
+  def checkConvergence(self,output):
+    '''
+      Determine convergence for Adaptive MonteCarlo
+      @ In, output, dictionary containing the results from Basic Statistic
```

Comment:
@ In, output, dictionary containing the results from Basic Statistic

->

@ In, output, dict,  dictionary containing the results from Basic Statistic